dan_layer/engine/Cargo.toml

```diff
@@ -19,6 +19,7 @@ tari_template_builtin = { workspace = true }
 tari_template_lib = { workspace = true }
 tari_utilities = { workspace = true }
 tari_transaction = { workspace = true }
+tari_consensus = { workspace = true }
```

Comment:
Consensus and execution must remain decoupled. 

I see you're using consensus constants to dictate the maximum permitted size of a template. 

I suggest creating a `TransactionProcessorConfig` struct that you pass into the processor that we can configure  using consensus constants in the validator/indexer's `bootstrap.rs`. The network can also go into that config.

```rust
pub struct TransactionProcessorConfig {
   pub network: Network,
  pub template_binary_max_size_bytes: usize
}
```

Reply:
okay, totally makes sense, let me update

Reply:
just pushed